### PR TITLE
Create categories for the source generators.

### LINKED ISSUE #165: Adding Categories?
Love the list, Source Generators are really neat.

But, what about creating categories?

There are about a dozen or so Dependency Injection Framworks in the list, and it would be really nice to have them in one place.

I could make a PR for that if there is interest in the idea.

### PR DESCRIPTION
**Pull request by FreeApophis**
Fixes #165

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,170 +22,298 @@ Add GitHub topic [`csharp-sourcegenerator`](https://github.com/topics/csharp-sou
...
+- [WrapperValueObject](https://github.com/martinothamar/WrapperValueObject) - ![stars](https://img.shields.io/github/stars/martinothamar/WrapperValueObject?style=flat-square&cacheSeconds=604800) ![last commit](https://img.shields.io/github/last-commit/martinothamar/WrapperValueObject?style=flat-square&cacheSeconds=86400) - for creating simple value objects wrapping primitive types.
+
+#### Immutability
+
+- [Immutype](https://github.com/DevTeam/Immutype) - ![stars](https://img.shields.io/github/stars/DevTeam/Immutype?style=flat-square&cacheSeconds=604800) ![last commit](https://img.shields.io/github/last-commit/DevTeam/Immutype?style=flat-square&cacheSeconds=86400) - generates extension methods to support immutability.
+- [Visor](https://github.com/leviysoft/Visor) - ![stars](https://img.shields.io/github/stars/leviysoft/Visor?style=flat-square&cacheSeconds=604800) ![last commit](https://img.shields.io/github/last-commit/leviysoft/Visor?style=flat-square&cacheSeconds=86400) C# optics library with incremental code generator (maintained fork of [suspended Tinkoff project](https://github.com/Tinkoff/Visor))
+
+#### Disriminated Unions
```
Comment (virzak) [inline README.md:124]: Discriminated

File: README.md [hunk 2]
```diff
@@ -22,170 +22,298 @@ Add GitHub topic [`csharp-sourcegenerator`](https://github.com/topics/csharp-sou
...
+<details>
+ <summary>Categories</summary>
+
+- [Dependency Injection (IoC Container)](#dependency-injection-ioc-container)
+- [Mappers](#mappers)
+- [Enums](#enums)
+- [Functional Programming](#functional-programming)
+  - [Value semantic / New Type Idiom](#value-semantic--new-type-idiom)
+  - [Immutability](#immutability)
+  - [Disriminated Unions](#disriminated-unions)
```
Comment (virzak) [inline README.md:40]: Discriminated

File: README.md [hunk 3]
```diff
@@ -22,170 +22,298 @@ Add GitHub topic [`csharp-sourcegenerator`](https://github.com/topics/csharp-sou
...
 - [SourceGeneratorPlayground](https://wengier.com/SourceGeneratorPlayground) - an online Source Generator Playground to play with generator ideas 💡 without any setup noise. [Source repo](https://github.com/davidwengier/SourceGeneratorPlayground).
-- [davidwengier/SourceGeneratorTemplate](https://github.com/davidwengier/SourceGeneratorTemplate)  - ![stars](https://img.shields.io/github/stars/davidwengier/SourceGeneratorTemplate?style=flat-square&cacheSeconds=604800) ![last commit](https://img.shields.io/github/last-commit/davidwengier/SourceGeneratorTemplate?style=flat-square&cacheSeconds=86400) A basic template for writing a C# source generator, from the Roslyn dev.
+- [davidwengier/SourceGeneratorTemplate](https://github.com/davidwengier/SourceGeneratorTemplate) -![stars](https://img.shields.io/github/stars/davidwengier/SourceGeneratorTemplate?style=flat-square&cacheSeconds=604800) ![last commit](https://img.shields.io/github/last-commit/davidwengier/SourceGeneratorTemplate?style=flat-square&cacheSeconds=86400) A basic template for writing a C# source generator, from the Roslyn dev.
 
 ## Source Generators
 
+
+
+<details>
```
Comment (skarllot) [inline README.md:31]: Need to be collapsed?

**Comment by skarllot** [comment]
Could be a category for Console/Command-line

File: README.md (hunk 1 above)
Comment (amis92) [reply README.md:124]:
```suggestion
#### Discriminated Unions
```

File: README.md (hunk 2 above)
Comment (amis92) [reply README.md:40]:
```suggestion
  - [Discriminated Unions](#discriminated-unions)
```

**Review by amis92 — COMMENTED** [review]
Overall looks good. I'm not sure if the subcategories that have only 1 or 2 items are necessary though. Opinions?

**Comment by skarllot** [comment]
I suggest the following changes regarding to the categories:
### Console/Command-line
- [AutoSpectre](https://github.com/jeppevammenkristensen/auto-spectre)
- [docopt.net](https://github.com/docopt/docopt.net)
- [Figgle](https://github.com/drewnoakes/figgle)

### Mappers
- [Flattening](https://github.com/Kros-sk/Kros.Generators.Flattening)
- [GraphQL.Tools](https://github.com/MoienTajik/GraphQL.Tools)

### Dependency Injection (IoC Container)
- [GrpcInjection](https://github.com/juniorporfirio/grpcinjection)
- [lambdajection](https://github.com/cythral/lambdajection)

### Communication
- [CoreWCF](https://github.com/CoreWCF/CoreWCF)
- [Imp.NET](https://github.com/DouglasDwyer/Imp.NET)
- [IoTHubClientGenerator](https://github.com/alonf/IoTHubClientGenerator)

### Graphics/Drawing
- [ComputeSharp](https://github.com/Sergio0694/ComputeSharp)
- [Svg to C# Source Generators](https://github.com/wieslawsoltes/Svg.Skia)

File: README.md (hunk 3 above)
Comment (FreeApophis) [reply README.md]: It's a space saver, I do not have an opinion either way:
Fixed in f590116bbb428469ca46ffde8d60ac8c0496bf51

**Review by amis92 — APPROVED** [review]
Perfect. Let's get it in, and we can work on enhancements in new PRs (if any).

**Review by skarllot — APPROVED** [review]

**Review by skarllot — APPROVED** [review]

**Comment by skarllot** [comment]
Will it be merged?

**Comment by FreeApophis** [comment]
It's all green :) someone just needs to merge.

**Pull request merged**